packages/react-button/src/components/Button/useButtonStyles.ts

```diff
@@ -63,160 +65,160 @@ const useRootStyles = makeStyles({
+      color: tokens.colorNeutralForegroundOnBrand,
     },
-  }),
-  subtle: theme => ({
-    backgroundColor: theme.colorSubtleBackground,
+  },
+  subtle: {
+    backgroundColor: tokens.colorSubtleBackground,
     ...shorthands.borderColor('transparent'),
-    color: theme.colorNeutralForeground2,
+    color: tokens.colorNeutralForeground2,
 
     ':hover': {
-      backgroundColor: theme.colorSubtleBackgroundHover,
+      backgroundColor: tokens.colorSubtleBackgroundHover,
       ...shorthands.borderColor('transparent'),
-      color: theme.colorNeutralForeground2BrandHover,
+      color: tokens.colorNeutralForeground2BrandHover,
     },
 
     ':active': {
-      backgroundColor: theme.colorSubtleBackgroundPressed,
+      backgroundColor: tokens.colorSubtleBackgroundPressed,
       ...shorthands.borderColor('transparent'),
-      color: theme.colorNeutralForeground2BrandPressed,
+      color: tokens.colorNeutralForeground2BrandPressed,
     },
-  }),
-  transparent: theme => ({
-    backgroundColor: theme.colorTransparentBackground,
+  },
+  transparent: {
+    backgroundColor: tokens.colorTransparentBackground,
     ...shorthands.borderColor('transparent'),
-    color: theme.colorNeutralForeground2,
+    color: tokens.colorNeutralForeground2,
 
     ':hover': {
-      backgroundColor: theme.colorTransparentBackgroundHover,
+      backgroundColor: tokens.colorTransparentBackgroundHover,
       ...shorthands.borderColor('transparent'),
-      color: theme.colorNeutralForeground2BrandHover,
+      color: tokens.colorNeutralForeground2BrandHover,
     },
 
     ':active': {
-      backgroundColor: theme.colorTransparentBackgroundPressed,
+      backgroundColor: tokens.colorTransparentBackgroundPressed,
       ...shorthands.borderColor('transparent'),
-      color: theme.colorNeutralForeground2BrandPressed,
+      color: tokens.colorNeutralForeground2BrandPressed,
     },
-  }),
+  },
 
   // Shape variations
-  circular: theme => ({
-    ...shorthands.borderRadius(theme.borderRadiusCircular),
-  }),
+  circular: {
+    ...shorthands.borderRadius(tokens.borderRadiusCircular),
+  },
   rounded: {
     /* The borderRadius rounded styles are handled in the size variations */
   },
-  square: theme => ({
-    ...shorthands.borderRadius(theme.borderRadiusNone),
-  }),
+  square: {
+    ...shorthands.borderRadius(tokens.borderRadiusNone),
+  },
 
   // Size variations
-  small: theme => ({
+  small: {
     ...shorthands.gap(buttonSpacing.smaller),
     ...shorthands.padding('0', buttonSpacing.medium),
 
     height: '24px',
     minWidth: '64px',
 
-    ...shorthands.borderRadius(theme.borderRadiusSmall),
+    ...shorthands.borderRadius(tokens.borderRadiusSmall),
 
-    fontSize: theme.fontSizeBase200,
-    fontWeight: theme.fontWeightRegular,
-    lineHeight: theme.lineHeightBase200,
-  }),
-  medium: theme => ({
+    fontSize: tokens.fontSizeBase200,
+    fontWeight: tokens.fontWeightRegular as React.CSSProperties['fontWeight'],
```

Comment:
It seems like either `MakeStylesStyles['fontWeight']` should be updated to allow css variables, or tokens.fontWeightRegular should include `as React.CSSProperties['fontWeight']`, so it doesn't need to be repeated everywhere it's used?

(This shouldn't block this PR)

Reply:
Agreed, I'll spin up a separate issue for this.

Reply:
@behowell I've noted this in #21212 and will work it out in a separate PR that addressed that issue.

Reply:
Here's the PR to fix the issue: #21217